### PR TITLE
PAT-2138 Risk profiler H2 upgrade plus tidying

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/dao/OcgmRepository.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ class OcgmRepository : DataRepository<OcgmList>() {
     data.fileName = filename
     data.fileType = FileType.OCGM
     data.reset()
-    csvData.stream().filter { p: List<String>? -> data.index.getAndIncrement() > 0 }
+    csvData.stream().filter { data.index.getAndIncrement() > 0 }
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/dao/OcgmRepository.kt:23]: Most of these changes are improving the kotlin which was auto-generated from java

File: build.gradle.kts [hunk 2]
```diff
@@ -18,8 +18,8 @@ val camelVersion = "3.14.1"
 dependencies {
   annotationProcessor("org.springframework.boot:spring-boot-configuration-processor")
 
-  runtimeOnly("com.h2database:h2")
-  runtimeOnly("org.flywaydb:flyway-core")
+  runtimeOnly("com.h2database:h2:2.1.210")
```
Comment (garethmdavies) [inline build.gradle.kts:21]: Could this be converted to testruntimeonly

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListener.kt [hunk 3]
```diff
@@ -51,15 +52,16 @@ class EventListener(
...
     try {
-      val message: Map<String, String?> = objectMapper.readValue(requestJson, Map::class.java) as Map<String, String?>
-      if (message == null || message["Message"] == null) {
+      val message: Map<String, Any?> = objectMapper.readValue(requestJson)
+      if (message["Message"] == null) {
         log.warn(requestJson)
       } else {
-        log.debug(message["Message"]) // do not log an excessive amount of data
-        event = objectMapper.readValue(message["Message"], OffenderEvent::class.java)
+        val body = message["Message"] as String
```
Comment (garethmdavies) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListener.kt:59]: Is there a chance of logging pii here or is it just the high level message event?

**Review by garethmdavies — APPROVED** [review]
LGTM , thanks for doing this change

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListener.kt (hunk 3 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/events/EventListener.kt:59]: sensitive info? I dont think so, its just incidents and alert info with codes , ids and nomsIds

File: build.gradle.kts (hunk 2 above)
Comment (steverendell) [reply build.gradle.kts:21]: good idea

**Pull request merged**